### ISSUE TITLE
MachinePool reconciliation get stuck when resourceGroup name has any capital letter in it 

### ISSUE THREAD

**Issue by primeroz** [kubernetes-sigs/cluster-api-provider-azure#2817]
/kind bug

While testing machinepools i ended up with a stuck cluster due to the reconciliation of the machinepool machines never completing and getting stuck in **ScaleSetScalingUp** state,  with the machines never reaching a ready state even though they are up and running form the point of view of the workload cluster. 

After lots of testing this turns out to be related to the `resourceGroup` name , when the name has any capital letter in it the machines are not found during the reconcile loop and so their status cannot be set to ready. See steps below to reproduce 

**What steps did you take and what happened:**
Complete steps to reproduce the issue below

With a running kind cluster with clusterctl initialized and the cluster-identity-secret already created

---
***Working Example using upstream machinepool flavour***

* Create workload cluster
```
export AZURE_CLUSTER_IDENTITY_SECRET_NAME=cluster-identity-secret
export AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE=default
export AZURE_CONTROL_PLANE_MACHINE_TYPE=Standard_D2s_v3
export AZURE_LOCATION=westeurope
export AZURE_NODE_MACHINE_TYPE=Standard_D2s_v3 
export AZURE_SUBSCRIPTION_ID=XXX
export AZURE_CLIENT_ID=XXX
export AZURE_TENANT_ID=XXX
export CLUSTER_IDENTITY_NAME=cluster-identity
export WORKER_MACHINE_COUNT=2

clusterctl generate cluster upstream --kubernetes-version v1.25.3 --flavor machinepool > /dev/shm/upstream.yaml

kubectl apply -f /dev/shm/upstream.yaml 
```

* Wait for cluster control plane to be available and deploy calico
```
clusterctl get kubeconfig upstream > /dev/shm/upstream

wait for first CP node

kubectl --kubeconfig=/dev/shm/upstream apply -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main/templates/addons/calico.yaml
```

* Wait for machinepool to get provisioned and nodes to join the cluster
**The Cluster is successfully provisioned and all reconciliation is successfull 
```
$ kubectl --kubeconfig /dev/shm/upstream get node      
NAME                           STATUS   ROLES           AGE    VERSION
upstream-control-plane-8nlqm   Ready    control-plane   3m5s   v1.25.3
upstream-mp-0000000            Ready    <none>          90s    v1.25.3
upstream-mp-0000001            Ready    <none>          82s    v1.25.3


$ clusterctl describe cluster upstream --echo --grouping=false

Cluster/upstream                                                         True                     2m16s           
├─ClusterInfrastructure - AzureCluster/upstream                          True                     4m48s           
├─ControlPlane - KubeadmControlPlane/upstream-control-plane              True                     2m16s           
│ └─Machine/upstream-control-plane-vx522                                 True                     2m17s           
│   ├─BootstrapConfig - KubeadmConfig/upstream-control-plane-mdmpx       True                     4m47s           
│   └─MachineInfrastructure - AzureMachine/upstream-control-plane-8nlqm  True                     2m17s           
└─Workers                                                                                                         
  └─MachinePool/upstream-mp-0                                            True                     17s             
    ├─BootstrapConfig - KubeadmConfig/upstream-mp-0                      True                     2m17s           
    └─MachineInfrastructure - AzureMachinePool/upstream-mp-0             True                     17s  

$ kubectl get azuremachinepoolmachine

NAME              VERSION   READY   STATE
upstream-mp-0-0   v1.25.3   true    Succeeded
upstream-mp-0-1   v1.25.3   true    Succeeded

$ kubectl get machinepool upstream-mp-0

NAME            CLUSTER    REPLICAS   PHASE     AGE     VERSION
upstream-mp-0   upstream   2          Running   7m23s   v1.25.3

$ kubectl get azuremachinepool upstream-mp-0
NAME            REPLICAS   READY   STATE
upstream-mp-0   2          true    Succeeded
```

* Delete cluster and wait for full cleanup 
```
kubectl delete cluster upstream
```

---
***NON Working Example using upstream machinepool flavour with a resourceGroup name with a capital letter in it***

* Change resourceGroup name and apply resources
```
➜ diff -u /dev/shm/upstream.yaml /dev/shm/upstream-rg-Upstream.yml
--- /dev/shm/upstream.yaml      2022-11-16 17:18:51.643233516 +0100
+++ /dev/shm/upstream-rg-Upstream.yml   2022-11-16 17:28:05.502390753 +0100
@@ -40,7 +40,7 @@
       role: node
     vnet:
       name: upstream-vnet
-  resourceGroup: upstream
+  resourceGroup: Upstream
   subscriptionID: a80dff0e-ab21-42a1-994a-f9285c4cfc92
 ---
 apiVersion: controlplane.cluster.x-k8s.io/v1beta1

$ kubectl apply -f /dev/shm/upstream-rg-Upstream.yml 
```

* Wait for cluster control plane to be available and deploy calico
```
clusterctl get kubeconfig upstream > /dev/shm/upstream

wait for first CP node

kubectl --kubeconfig=/dev/shm/upstream apply -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main/templates/addons/calico.yaml
```

* Wait for machinepool to get provisioned and nodes to join the cluster
**The Cluster is NOT successfully provisioned , azuremachinepoolmachine resources never reach a ready state, machine pool is stuck in ScaleSetScalingUp**

```
➜ kubectl --kubeconfig /dev/shm/upstream get node 
NAME                           STATUS   ROLES           AGE     VERSION
upstream-control-plane-t7ms4   Ready    control-plane   5m14s   v1.25.3
upstream-mp-0000000            Ready    <none>          2m51s   v1.25.3
upstream-mp-0000001            Ready    <none>          2m52s   v1.25.3

➜ clusterctl describe cluster upstream --echo --grouping=false
NAME                                                                     READY  SEVERITY  REASON             SINCE  MESSAGE 
Cluster/upstream                                                         True                                5m              
├─ClusterInfrastructure - AzureCluster/upstream                          True                                7m18s           
├─ControlPlane - KubeadmControlPlane/upstream-control-plane              True                                5m              
│ └─Machine/upstream-control-plane-w5cm4                                 True                                5m2s            
│   ├─BootstrapConfig - KubeadmConfig/upstream-control-plane-ffm5n       True                                7m17s           
│   └─MachineInfrastructure - AzureMachine/upstream-control-plane-t7ms4  True                                5m3s            
└─Workers                                                                                                                    
  └─MachinePool/upstream-mp-0                                            False  Info      ScaleSetScalingUp  2m46s           
    ├─BootstrapConfig - KubeadmConfig/upstream-mp-0                      True                                4m47s           
    └─MachineInfrastructure - AzureMachinePool/upstream-mp-0             False  Info      ScaleSetScalingUp  2m46s  


➜ kubectl get azuremachinepoolmachine
NAME              VERSION   READY   STATE
upstream-mp-0-0                     Succeeded
upstream-mp-0-1                     Succeeded


➜  kubectl get machinepool upstream-mp-0

NAME            CLUSTER    REPLICAS   PHASE          AGE     VERSION
upstream-mp-0   upstream              Provisioning   9m22s   v1.25.3


➜ kubectl get azuremachinepool upstream-mp-0
NAME            REPLICAS   READY   STATE
upstream-mp-0                      Updating
```

* Machines never get into Ready / Healthy State 
```
 λ kubectl get azuremachinepoolmachine.infrastructure.cluster.x-k8s.io/upstream-mp-0-0 -o yaml | yq .status
{
  "conditions": [
    {
      "lastTransitionTime": "2022-11-16T16:36:18Z",
      "reason": "NodeProvisioning",
      "severity": "Info",
      "status": "False",
      "type": "Ready"
    },
    {
      "lastTransitionTime": "2022-11-16T16:36:18Z",
      "reason": "NodeProvisioning",
      "severity": "Info",
      "status": "False",
      "type": "NodeHealthy"
    }
  ],
  "latestModelApplied": true,
  "provisioningState": "Succeeded"
}
```

**What did you expect to happen:**
Provisioning of machinepools should work with any name that is valid for resourceGroups. 

**Anything else you would like to add:**

I have been reusing the same name for the resourceGroup during my testing including before starting to test machinepools.

With MachineDeployments i had no issues related to the resourceGroup name

**Environment:**

- cluster-api-provider version: 1.2.5
- cluster-api-provider-azure version: 1.6.0
- Kubernetes version: (use `kubectl version`): Management: 1.23.12 , Workload : 1.25.3
- OS (e.g. from `/etc/os-release`):

**Comment by primeroz** [comment]
I could find code to actually manage the conversion of `resourceGroupName` to lower case so i guess maybe is just missing somewhere. 

I will try and have a look around 

```
➜ rg ConvertResourceGroupNameToLower
exp/controllers/azuremanagedmachinepool_reconciler.go
137:            providerID, err := azureutil.ConvertResourceGroupNameToLower(azure.ProviderIDPrefix + *instances[i].ID)
                                                                                                                                                                                                                   
azure/services/virtualmachines/virtualmachines.go
100:            providerID, err := azureutil.ConvertResourceGroupNameToLower(azure.ProviderIDPrefix + infraVM.ID)
                                                    
azure/services/scalesets/scalesets.go
110:                    providerID, err := azureutil.ConvertResourceGroupNameToLower(azure.ProviderIDPrefix + fetchedVMSS.ID)                                      
                                                                                                         
util/azure/azure_test.go                           
25:func TestConvertResourceGroupNameToLower(t *testing.T) {
76:                     actual, err = ConvertResourceGroupNameToLower(test.resourceID)                 
                                                    
util/azure/azure.go
30:// ConvertResourceGroupNameToLower converts the resource group name in the resource ID to be lowered.
32:func ConvertResourceGroupNameToLower(resourceID string) (string, error) {

```

**Comment by CecileRobertMichon** [comment]
This was fixed in #2894 

/close
